internal/backend/user.go

```diff
@@ -53,12 +53,12 @@ func newUser(ctx context.Context, userID string, db *DB, remote *remote.User, st
 		labels := pprof.Labels("go", "Connector Updates", "UserID", user.userID)
 		pprof.Do(ctx, labels, func(_ context.Context) {
 			for update := range remote.GetUpdates() {
-				update := update
-				if err := user.tx(context.Background(), func(tx *ent.Tx) error {
+				update := func() error {
 					defer update.Done()
-					return user.apply(context.Background(), tx, update)
-				}); err != nil {
-					logrus.WithError(err).Errorf("Failed to apply update: %v", update)
+					return user.apply(context.Background(), update)
+				}
+				if err := update(); err != nil {
+					logrus.WithError(err).Errorf("Failed to apply update: %v", err)
 				}
```

Comment:
now that `user.apply` is only used for external updates, we should probably just move this deferred `update.Done()` call into `user.apply` itself. That would simplify this weirdness here.

+ the docstring on `user.apply` is out of date, there's no `source` anymore.

Reply:
Fixed.